File: backend/components/api/wire_types/src/resources/annotation.rs
use crate::Actor;
use crate::Apiv2Response;
use crate::DateTime;
use crate::Serialize;
use crate::Utc;
use newtypes::AnnotationId;

#[derive(Debug, Clone, Serialize, Apiv2Response, macros::JsonResponder)]
pub struct Annotation {
    pub id: AnnotationId,
    pub timestamp: DateTime<Utc>,
    pub note: String,
    pub is_pinned: bool,
    pub source: Actor,
}

File: frontend/packages/types/src/data/annotation.ts
import type { ActorApiKey, ActorFirmEmployee, ActorFootprint, ActorOrganization } from './actor';

export type AnnotationSource = ActorFootprint | ActorOrganization | ActorFirmEmployee | ActorApiKey;

export type Annotation = {
  id: string;
  note: string;
  isPinned: boolean;
  source: AnnotationSource;
  timestamp: string;
};
